torch/fx/graph.py

```diff
@@ -455,8 +470,27 @@ def append_stacktrace_summary(node : Node):
                     prev_stacktrace = ""
                     body.append('\n# No stacktrace found for following nodes \n')
 
+        def stringify_shape(shape : torch.Size) -> str:
+            return f"[{', '.join(str(x) for x in shape)}]"
+
         def emit_node(node : Node):
             maybe_type_annotation = '' if node.type is None else f' : {type_repr(node.type)}'
+
+            if verbose:
+                # override annotation with more detailed information
+                from torch._subclasses.fake_tensor import FakeTensor
+                from torch.fx.experimental.symbolic_shapes import PySymInt
+                from torch.fx.passes.shape_prop import TensorMetadata
+
+                meta_val = node.meta.get('val', node.meta.get('tensor_meta', None))
+
+                if isinstance(meta_val, FakeTensor):
+                    maybe_type_annotation = f': Tensor<{dtype_abbrs[meta_val.dtype]}>{stringify_shape(meta_val.shape)}'
+                elif isinstance(meta_val, PySymInt):
```

Comment:
update to include PySymFloat too

see https://github.com/pytorch/pytorch/pull/86425/files#r991657516 for some infra to leverage